lib/pull_preview/instance.rb

```diff
@@ -12,12 +12,55 @@ class Instance
     attr_reader :subdomain
     attr_reader :ports
     attr_reader :registries
+    attr_reader :ip_prefix
+    attr_reader :swap_enabled
+    alias swap_enabled? swap_enabled
 
     class << self
       attr_accessor :client
       attr_accessor :logger
     end
 
+    class Provisioner
+      COMPOSE_VERSION = "1.29.2".freeze
+
+      class << self
+        def ssh_access(ssh_public_key)
+          %{echo '#{ssh_public_key}' > /home/ec2-user/.ssh/authorized_keys}
+        end
+
+        def prepare_user(remote_app_path)
+          [
+            "mkdir -p #{remote_app_path} && chown -R ec2-user.ec2-user #{remote_app_path}",
+            "echo 'cd #{remote_app_path}' > /etc/profile.d/pullpreview.sh",
+            "echo '[[ -f /etc/pullpreview/env ]] && set -o allexport; source /etc/pullpreview/env; set +o allexport' >> /etc/profile.d/pullpreview.sh"
+          ]
+        end
+
+        def setup_swapping
+          [
+            "sudo sed -i '/^tmpfs/c\tmpfs       \/dev\/shm    tmpfs   defaults,size=256M  0   0' /etc/fstab",
+            "echo 'vm.vfs_cache_pressure=50' | tee -a /etc/sysctl.conf"
+          ]
+        end
```

Comment:
This is still incomplete. should be...
```suggestion
        def setup_swapping
          [
            "fallocate -l 2G /swapfile && chmod 600 /swapfile && mkswap /swapfile && swapon /swapfile",
            "echo '/swapfile none swap sw 0 0' | tee -a /etc/fstab",
            "sysctl vm.swappiness=10 && sysctl vm.vfs_cache_pressure=50",
            "echo 'vm.swappiness=10' | tee -a /etc/sysctl.conf",
            "echo 'vm.vfs_cache_pressure=50' | tee -a /etc/sysctl.conf"
          ]
        end
```